File: go/benchmark.go
package main

import (
    "fmt"
    "math"
    "os"
    "time"
)

func isPrime(n int) bool {
    if n <= 1 {
        return false
    }
    for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
        if n%i == 0 {
            return false
        }
    }
    return true
}

func main() {
    start := time.Now()

    count := 0
    num := 2
    for count < 100000 {
        if isPrime(num) {
            count++
        }
        num++
    }

    elapsed := time.Since(start).Seconds()

    file, _ := os.OpenFile("benchmark_results.csv", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
    defer file.Close()
    file.WriteString(fmt.Sprintf("Go,%.2f\n", elapsed))

}

File: c/benchmark.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>
#include <time.h>

bool is_prime(int n) {
    if (n <= 1) return false;
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    clock_t start = clock();

    int count = 0;
    int num = 2;
    while (count < 100000) {
        if (is_prime(num)) {
            count++;
        }
        num++;
    }

    clock_t end = clock();
    double elapsed_time = (double)(end - start) / CLOCKS_PER_SEC;

    FILE *fp = fopen("benchmark_results.csv", "a");
    fprintf(fp, "C,%.2f\n", elapsed_time);
    fclose(fp);

    return 0;
}
